p2p/netaddress.go

```diff
@@ -32,8 +32,13 @@ type NetAddress struct {
 	str string
 }
 
-// IDAddressString returns id@hostPort.
+// IDAddressString returns {protocol://}ID@host:port.
 func IDAddressString(id ID, hostPort string) string {
+	// we respect the protocol definition in here.
```

Comment:
We merged a similar fix in https://github.com/tendermint/tendermint/pull/2797/files#diff-91e8069d5cb91a5b3067a01482559679R38. There we just drop the protocol prefix, since we shouldn't need it anymore. Sorry didn't see this earlier, it would have saved us some time.